nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2226,27 +2261,30 @@ impl<'a> Planner<'a> {
             .map_err(|_| Error::NoActiveNexusZonesInParentBlueprint)
     }
 
-    // Returns if the zone is ready to be updated.
+    // Returns whether the out-of-date Nexus zone is ready to be updated.
+    //
+    // For reporting purposes, we assume that we want the supplied
+    // zone to be expunged or updated because it is out-of-date.
     //
     // If the zone should not be updated yet, updates the planner report to
     // identify why it is not ready for update.
-    fn can_zone_be_updated(
+    //
+    // Precondition: zone must be a Nexus zone
```

Comment:
```suggestion
    // Precondition: zone must be a Nexus zone and be running an out-of-date image
```

(trying to communicate that this shouldn't be used in any case other than "update")